### PR TITLE
arc_unpacker: 2019-01-28 -> 2021-05-17, add darwin build

### PR DESCRIPTION
**Pull request by futile**
###### Motivation for this change

Upstream update makes us no longer require the patch. Also add a darwin build based on @midchildan's diff from https://github.com/NixOS/nixpkgs/pull/123091#issuecomment-841681592.

@midchildan ping; I don't have a darwin system, but noticed that hydra didn't build a darwin version, so I added it as a platform. Also mentioned you as a co-author for the commit. Is this in line with what you suggested in the comment?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>arc_unpacker</li>
  </ul>
</details>

### CODE REVIEW

**Comment by futile** [comment]
This seems to fail on aarch64-linux, which I didn't specifically intend to add. But no build on darwin?

**Review by midchildan — CHANGES_REQUESTED** [review]

**Comment by midchildan** [comment]
The failure may be due to a problem with the test. Maybe add a `doCheck = !stdenv.isAarch64` for now?

**Comment by midchildan** [comment]
The Darwin build worked like a charm.

File: pkgs/tools/archivers/arc_unpacker/default.nix [hunk 1]
```diff
@@ -58,5 +52,6 @@ stdenv.mkDerivation rec {
     homepage = "https://github.com/vn-tools/arc_unpacker";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ midchildan ];
+    platforms = platforms.linux ++ platforms.darwin;
```
[the remark this answers is not in the document]
Comment (futile) [reply pkgs/tools/archivers/arc_unpacker/default.nix]: Added that.

**Comment by futile** [comment]
> The aarch64 failure may be due to a problem with the test. Maybe add a `doCheck = !stdenv.isAarch64` for now?

Done.

> The Darwin (x86_64) build worked like a charm.

Awesome! Did you test it locally or is that somehow visible from the checks on github? Only

```
arc_unpacker, arc_unpacker.passthru.tests on aarch64-linux
arc_unpacker, arc_unpacker.passthru.tests on x86_64-linux
```

looked like builds on those platforms to me, so I thought no build on darwin happened.

**Comment by midchildan** [comment]
I tested locally on my Macbook.

**Review by midchildan — APPROVED** [review]

**Comment by futile** [comment]
> I tested locally on my Macbook.

Ah okay, thanks! :)

**Comment by midchildan** [comment]
This should be part of ZHF as it fixes the Darwin build. Although there's a version bump, there are no source code differences as it's just an upstreaming of a patch.

ZHF: #122042

ping @NixOS/nixos-release-managers

**Review by midchildan — APPROVED** [review]

File: pkgs/tools/archivers/arc_unpacker/default.nix [hunk 2]
```diff
@@ -51,12 +45,14 @@ stdenv.mkDerivation rec {
...
-  doCheck = true;
+  # A few tests fail on aarch64
+  doCheck = !stdenv.isAarch64;
 
   meta = with lib; {
     description = "A tool to extract files from visual novel archives";
     homepage = "https://github.com/vn-tools/arc_unpacker";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ midchildan ];
+    platforms = platforms.unix ++ platforms.windows;
```
Comment (SuperSandro2000) [inline pkgs/tools/archivers/arc_unpacker/default.nix:56]:
```suggestion
    platforms = platforms.all;
```

Comment (futile) [reply pkgs/tools/archivers/arc_unpacker/default.nix]: done

File: pkgs/tools/archivers/arc_unpacker/default.nix [hunk 3]
```diff
@@ -1,29 +1,23 @@
-{ lib, stdenv, fetchFromGitHub, cmake, makeWrapper, boost, libpng, libjpeg, zlib
-, openssl, libwebp, catch }:
+{ lib, stdenv, fetchFromGitHub, cmake, makeWrapper, boost, libpng, libiconv
+, libjpeg, zlib, openssl, libwebp, catch }:
 
 stdenv.mkDerivation rec {
   pname = "arc_unpacker-unstable";
```
Comment (SuperSandro2000) [inline pkgs/tools/archivers/arc_unpacker/default.nix:5]:
```suggestion
  pname = "arc_unpacker";
```

File: pkgs/tools/archivers/arc_unpacker/default.nix [hunk 4]
```diff
@@ -1,29 +1,23 @@
...
-  version = "2019-01-28";
+  version = "2021-05-17";
```
Comment (SuperSandro2000) [inline pkgs/tools/archivers/arc_unpacker/default.nix:6]:
```suggestion
  version = "unstable-2021-05-17";
```

File: pkgs/tools/archivers/arc_unpacker/default.nix (hunk 3 above)
Comment (midchildan) [reply pkgs/tools/archivers/arc_unpacker/default.nix]: I believe the official policy is to put "unstable" in the name per the [documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/contributing/coding-conventions.chapter.md#package-naming-sec-package-naming):

> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

I'm assuming that "name" in this context means `pname`.

**Pull request merged**